### PR TITLE
Allow passing options to `qs.stringify`

### PR DESCRIPTION
**Pull request by jakewtaylor**
# Why

I need to change how the query parameters are stringified 😄

### CODE REVIEW

**Comment by jakewtaylor** [comment]
Our API follows a pattern like this:
`/endpoint?key=value1&key=value2&key=value3`

I'd like to be able to pass in `qs` options, so I can control how the array is encoded - from their docs:
```JavaScript
qs.stringify({ a: ['b', 'c'] }, { arrayFormat: 'indices' })
// 'a[0]=b&a[1]=c'
qs.stringify({ a: ['b', 'c'] }, { arrayFormat: 'brackets' })
// 'a[]=b&a[]=c'
qs.stringify({ a: ['b', 'c'] }, { arrayFormat: 'repeat' })
// 'a=b&a=c'
qs.stringify({ a: ['b', 'c'] }, { arrayFormat: 'comma' })
// 'a=b,c'
```

I would need 'repeat' 😄

**Comment by jakewtaylor** [comment]
Maybe a better option would be to let users pass in an `encodeQueryParams` method to the provider, to let users handle it themselves.

**Comment by jakewtaylor** [comment]
I'll make another PR that goes down the `encodeQueryParams` method route, I'll try get it done later tonight. 😄

File: src/useGet.tsx [hunk 1]
```diff
@@ -24,6 +24,10 @@ export interface UseGetProps<TData, TQueryParams> {
    * Query parameters
    */
   queryParams?: TQueryParams;
+  /**
+   * Query parameter stringify options
+   */
+  queryParamStringifyOptions?: IStringifyOptions;
```
[the remark this answers is not in the document]
Comment (fabien0102) [reply src/useGet.tsx]: Let's go for a `queryParamStringify?: (obj: any) => string` with `qs.stringify` as default.

So people can escape as they want if needed

**Comment by TejasQ** [comment]
@jakewtaylor 

# you are a champion

champion

**Comment by jakewtaylor** [comment]
I got started on this last night, but I was having issues with typescript - will continue on when I get the chance. 😄

**Comment by TejasQ** [comment]
Happy to support you with any help you need, @jakewtaylor!

**Comment by jakewtaylor** [comment]
Sorry guys - haven't had much time to do this.. 😢

**Comment by TejasQ** [comment]
@jakewtaylor are you able to find time for this? If not, that's OK – I'd be able to cover the outstanding work here.

**Comment by jakewtaylor** [comment]
Probably not, to be honest. Feel free to take over 😄

**Comment by fabien0102** [comment]
Closed in favor of #240 (same branch rebase and finished)

**Pull request closed without merging**